src/Microsoft.HttpRepl.Tests/Commands/DeleteCommandsTest.cs

```diff
@@ -0,0 +1,107 @@
+    {
+        private readonly DeleteCommandsConfig _config;
+        public DeleteCommandTests(DeleteCommandsFixture DeleteCommandsFixture)
+        {
+            _config = DeleteCommandsFixture.Config;
+        }
+
+        [Fact]
+        public async Task ExecuteAsync_WithNoBasePath_VerifyError()
+        {
+            HttpState httpState = new HttpState();
+
+            string expectedErrorMessage = Resources.Strings.Error_NoBasePath.SetColor(httpState.ErrorColor);
+            string actualErrorMessage = null;
+
+            IShellState shellState = MockHelpers.GetMockedShellState(errorMessageCallback: (s) => actualErrorMessage = s);
+
+            DeleteCommand command = new DeleteCommand();
+            ICoreParseResult parseResult = CoreParseResultHelper.Create("DELETE");
+
+            await command.ExecuteAsync(shellState, httpState, parseResult, CancellationToken.None);
+
+            Assert.Equal(expectedErrorMessage, actualErrorMessage);
+        }
+
+        [Fact]
+        public async Task ExecuteAsync_WithMultipartRoute_VerifyOutput()
+        {
+            int expectedResponseLines = 5; // The path, three headers and the response.
+            string expectedResponseContent = "File path delete received successfully.";
+            List<string> actual = new List<string>();
+
+            IShellState shellState = MockHelpers.GetMockedShellState(writeLineCallback: (s) => actual.Add(s));
+
+            HttpState httpState = HttpStateHelpers.Create(_config.BaseAddress, "a/file/path.txt");
+
+            DeleteCommand command = new DeleteCommand();
+            ICoreParseResult parseResult = CoreParseResultHelper.Create("DELETE");
+
+            await command.ExecuteAsync(shellState, httpState, parseResult, CancellationToken.None);
+
+            Assert.Equal(expectedResponseLines, actual.Count);
+            Assert.Equal(expectedResponseContent, actual[expectedResponseLines - 1]);
+        }
+
+        [Fact]
+        public async Task ExecuteAsync_WithOnlyBaseAddress_VerifyOutput()
+        {
+            int expectedResponseLines = 5; // The path, three headers and the response.
+            string expectedResponseContent = "Root delete received successfully.";
+            List<string> actual = new List<string>();
+
+            IShellState shellState = MockHelpers.GetMockedShellState(writeLineCallback: (s) => actual.Add(s));
+
+            HttpState httpState = HttpStateHelpers.Create(_config.BaseAddress);
+
+            DeleteCommand command = new DeleteCommand();
+            ICoreParseResult parseResult = CoreParseResultHelper.Create("DELETE");
+
+            await command.ExecuteAsync(shellState, httpState, parseResult, CancellationToken.None);
+
+            Assert.Equal(expectedResponseLines, actual.Count);
+            Assert.Equal(expectedResponseContent, actual[expectedResponseLines - 1]);
+        }
+    }
+
+    public class DeleteCommandsFixture : IDisposable
```

Comment:
These fixtures are all the same but for the Config type (as far as I can tell).  Could we have something like this to consolidate them?
```
class TestCommandFixture<TConfigType> 
    where TConfigType : SampleApiServerConfig
```
Then the tests would be something like:
```
public class DeleteCommandTests : IClassFixture<TestCommandFixture<DeleteCommandsConfig>>
```

Reply:
Good call. Done.